gunrock/app/bc/bc_enactor.cuh

```diff
@@ -602,14 +599,16 @@ public:
     }
 
     /**
-      * @brief one run of sssp, to be called within GunrockThread
+      * @brief one run of BC, to be called within GunrockThread
       * @param thread_data Data for the CPU thread
       * \return cudaError_t error message(s), if any
       */
     cudaError_t Run(ThreadSlice &thread_data)
     {
-        gunrock::app::Iteration_Loop<0, 1, ForwardIterationT>(thread_data, forward_iterations[thread_data.thread_num]);
-        gunrock::app::Iteration_Loop<0, 1, BackwardIterationT>(thread_data, backward_iterations[thread_data.thread_num]);
+        gunrock::app::Iteration_Loop<0, 1, ForwardIterationT>(
+            thread_data, forward_iterations[thread_data.thread_num]);
+        gunrock::app::Iteration_Loop<0, 2, BackwardIterationT>(
```

Comment:
What do the `0` and `2` mean here?

Reply:
The first number (0) is the number of associative values of type VertexT, for each vertex in the remote sub-frontier, to be communicated in multi-GPU setup. The second number (1 for forward, and 2 for backward), are the number of associative values of type ValueT, for each remote vertex. 
`max_num_vertex_associates` and `max_num_value__associates` should be set to the maximum of all iteration loops, so `EnactorBase::Init()` can allocate the suitable amount of memory for communication